ee/src/main/java/org/jboss/as/ee/concurrent/ControlPointUtils.java

```diff
@@ -184,8 +188,15 @@ public void run() {
                         throw EeLogger.ROOT_LOGGER.cannotRunScheduledTask(runnable);
                     }
                 } catch (Exception e) {
-                    EeLogger.ROOT_LOGGER.failedToRunTask(e);
+                    // WFLY-13043
```

Comment:
I don't think any of this exception handling logic is necessary.  If you just get rid of this try/catch block entirely, then any runtime exceptions from run() will propagate to the underlying executor service.

Reply:
It would, if for some reason beginRequest() did not indicate it throws Exception ( which it does not in any part of impl). Yeah, I dont like this code, but unless controlPoint change API it has to be like that.

Reply:
Agreed - ControlPoint has an unfortunate method signature...
However, you can still simplify this code to be more readable, e.g.

```java
try {
    if (controlPoint.beginRequest() == RunResult.RUN) {
        try {
            runnable.run();
        } catch (RuntimeException e) {
            throw new ExecutionException(e);
        } finally {
            controlPoint.requestComplete();
        }
    }
} catch (ExecutionException e) {
    throw EeLogger.ROOT_LOGGER.failureWhileRunningTask(runnable, e.getCause());
} catch (Exception e) {
    EeLogger.ROOT_LOGGER.failedToRunTask(runnable,e);
}
```

Reply:
I guess, though I just dont like to re-throw exception few lines down the pipe